src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -67,12 +67,18 @@ class Crystal::Doc::Generator
     end
 
     if filename
-      body = doc(program_type, File.read(filename))
+      raw_body = File.read(filename)
+      body = doc(program_type, raw_body)
     else
+      raw_body = ""
       body = ""
     end
 
     File.write "#{@dir}/index.html", MainTemplate.new(body, types, repository_name)
+
+    main_index = Main.new(raw_body, Type.new(self, @program), repository_name)
+    File.write "#{@dir}/index.json", main_index
+    File.write "#{@dir}/index.jsonp", main_index.to_jsonp
```

Comment:
I don't think JSONP support is needed because JSONP is the stone age hack and people living in civilization have XHR2 which has better cross domain support over CORS. And Good News, GitHub Pages now supports `Access-Control-Allow-Origin: *`, so we can host JSON file by GitHub Pages to other domains. Why do you need JSONP?